### ISSUE TITLE
database keeps absolute path to project.yaml references/files

### ISSUE THREAD

**Issue by thatONEjustin** [craftcms/cms#3546]
### Description
If `useProjectConfigFile` is set to `true`, the paths for:

- `dateModified`
- `email`
- `fieldGroups`
- `fields`
- `plugins`
- `sections`
- `siteGroups`
- `sites`
- `system`
- `users`
- `volumes`
- `matrixBlockTypes`

are all hardcoded into the database. This is causing an issue between team members as they get file permission issues for a path thaat doesn't exist. 

### Steps to reproduce

1. Install Craft 3.1 Beta
2. set `useProjectConfigFile` to `true` in `config/general.php`
3. export database (using whatever tools you need)

### Example from my export
```
{\"dateModified\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"email\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"fieldGroups\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"fields\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"plugins\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"sections\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"siteGroups\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"sites\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"system\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"users\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"volumes\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\",\"matrixBlockTypes\":\"/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml\"}');
```

### Additional info

- Craft version: `"craftcms/cms": "3.1.0-beta.3"`
- PHP version: 7.1

**Comment by brandonkelly** [comment]
@andris-sevcenko We may be able to avoid storing absolute paths by passing them through `Craft::alias()`, which would convert `/Users/justin/Desktop/wojo/projects/eregulations/config/project.yaml` to `@config/project.yaml`, for example.

But then there’s the question of what the desired behavior should be when a DB is improved into another environment. Maybe it’s best that the paths are absolute, and if the files don’t exist we just assume all of the files could have been modified, sync everything, and update the config map.

**Comment by andris-sevcenko** [comment]
As far as config map goes, it's there just so when you save anything to project config, it's known to which file to write the changes to, so storing absolute paths isn't required. 

If for some reason, DB from another environment is brought in with different values in the config map, then Craft will just store the respective nodes in those files.

As far as modified files go, those are cached using `Craft::$app->getCache()` so aren't affected. If a file that has a cached modified time goes missing, a sync is triggered.

**Comment by andris-sevcenko** [comment]
Closed via 5ccd5ff
